utilities.tex

```diff
@@ -124,13 +116,67 @@
 \setcounter{subsection}{1}
 \rSec2[utility.swap]{swap}
 
-\ednote{The following edits to the two \tcode{swap} overloads are to be applied to the versions in
-namespace \tcode{std}.}
-
+{\color{newclr}
 \indexlibrary{\idxcode{swap}}%
+\pnum The name \tcode{swap} denotes a customization point
+object~(\ref{customization.point.object}). The effect of the expression
+\tcode{ranges::swap(E1, E2)} for some expressions \tcode{E1}
+and \tcode{E2} is equivalent to:
+
+\begin{itemize}
+\item
+  \tcode{(void)swap(E1, E2)}, with overload resolution
+  performed in a context that includes the declarations
+\begin{codeblock}
+  template <class T>
+  void swap(T&, T&) = delete;
+  template <class T, size_t N>
+  void swap(T(&)[N], T(&)[N]) = delete;
+\end{codeblock}
+  and does not include a declaration of \tcode{ranges::swap}.
+  If the function selected by overload resolution does not
+  exchange the values denoted by \tcode{E1} and \tcode{E2},
+  the program is ill-formed with no diagnostic required.
+
+\item
+  Otherwise, \tcode{(void)swap_ranges(E1, E2)} if \tcode{E1} and
+  \tcode{E2} are lvalues of array types~(\cxxref{basic.compound})
+  of equal extent and \tcode{ranges::swap(*(E1), *(E2))}
+  is a valid expression, except that
+  \tcode{noexcept(ranges::swap(E1, E2))} is equal to
+  \tcode{noexcept(ranges::swap(*(E1), *(E2)))}.
```

Comment:
You have generalized this to include swapping arrays of different type. I know that's your ultimate intent, but perhaps this deserves an \ednote stating that this generalization was intentional. It deserves review discussion.